org.spoofax.interpreter.core/src/main/java/org/spoofax/interpreter/library/ssl/StrategoImmutableRelation.java

```diff
@@ -0,0 +1,93 @@
+package org.spoofax.interpreter.library.ssl;
+
+import io.usethesource.capsule.BinaryRelation;
+
+import org.spoofax.interpreter.terms.IStrategoTerm;
+import org.spoofax.interpreter.terms.ITermFactory;
+import org.spoofax.interpreter.terms.ITermPrinter;
+import org.spoofax.interpreter.util.EntryAsPairIterator;
+import org.spoofax.terms.StrategoTerm;
+import org.spoofax.terms.TermFactory;
+import java.io.IOException;
+import java.util.Iterator;
+
+import static org.spoofax.terms.AbstractTermFactory.EMPTY_TERM_ARRAY;
+
+public class StrategoImmutableRelation extends StrategoTerm implements IStrategoTerm {
+    public final BinaryRelation.Immutable<IStrategoTerm, IStrategoTerm> backingRelation;
+
+    public StrategoImmutableRelation(BinaryRelation.Immutable<IStrategoTerm, IStrategoTerm> backingRelation) {
+        super(TermFactory.EMPTY_LIST);
+        this.backingRelation = backingRelation;
+    }
+
+    public StrategoImmutableRelation() {
+        this(BinaryRelation.Immutable.of());
+    }
+
+    @SuppressWarnings("NullableProblems")
+    @Override
+    public Iterator<IStrategoTerm> iterator() {
+        return new EntryAsPairIterator(backingRelation.entryIterator());
+    }
+
+    @Override
+    public int getSubtermCount() {
+        return 0;
+    }
+
+    @Override
+    public IStrategoTerm getSubterm(int index) {
+        throw new UnsupportedOperationException();
```

Comment:
Should this be `IndexOutOfBoundsException`?

Reply:
Yes, to have parity of other term implementations it should probably be index of out bounds.